src/test/java/com/example/spring/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,100 @@
+package com.example.spring.controller;
+
+import static org.junit.Assert.assertNotNull;
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.example.spring.dto.ShoppingCartDto;
+import com.example.spring.security.JwtUtil;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@Sql(
+        scripts = {
+                "classpath:database/shopping.carts/clear-cart.sql",
+                "classpath:database/cart.items/clear-cart-items.sql",
+                "classpath:database/books/clear-books.sql",
+                "classpath:database/shopping.carts/add-cart.sql",
+                "classpath:database/roles/add-role.sql",
+                "classpath:database/users-roles/add-user-role.sql",
+                "classpath:database/books/add-book.sql",
+                "classpath:database/cart.items/add-cart-items.sql",
+        },
+        executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD
+)
+@Sql(
+        scripts = {
+                "classpath:database/users-roles/clear-users-roles.sql",
+                "classpath:database/roles/clear-role.sql",
+                "classpath:database/cart.items/clear-cart-items.sql",
+                "classpath:database/shopping.carts/clear-cart.sql",
+                "classpath:database/books/clear-books.sql"
+        },
+        executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+)
+public class ShoppingCartControllerTest {
```

Comment:
I see that you have more than two controllers in this class. Let's test others ;)